File: Medium/209-minimum-size-subarray-sum/try-3.cpp
/*
 * Problem #209: Minimum Size Subarray Sum
 * Difficulty: Medium
 * Submission: Try 3
 * status: Accepted
 * Language: cpp
 * Date: 08/12/2024, 17:43:19
 * Link: https://leetcode.com/problems/minimum-size-subarray-sum/
 */

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int summ=0,i=0,mx=INT_MAX;
        for(int e=0;e<nums.size();e++){
            summ+=nums[e];
            while(summ>=target){
                mx=min(mx,e-i+1);
                summ-=nums[i];
                i++;
            }
        }
        return (mx==INT_MAX?0:mx);
    }
};

File: Medium/238-product-of-array-except-self/try-3.py
/*
 * Problem #238: Product of Array Except Self
 * Difficulty: Medium
 * Submission: Try 3
 * status: Accepted
 * Language: python3
 * Date: 14/12/2024, 19:40:35
 * Link: https://leetcode.com/problems/product-of-array-except-self/
 */

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        l1=[1]
        p=1

        for i in range(len(nums)-1):
            p=p*nums[i]
            l1.append(p)
        p=1
        
        for i in range(len(nums)-1,-1,-1):
            l1[i]=l1[i]*p
            p=p*nums[i]

        return l1
            



        
